docs/development/development.md

```diff
@@ -0,0 +1,258 @@
+- **`github.com/Masterminds/sprig`**: This library provides a
+
+set of additional functions for Go templates, extending the functionality of the standard template engine. It’s used for template rendering, which allows the project to handle dynamic configuration files with more complex logic, such as string manipulations and formatting.
+
+- **`k8s.io/client-go`**: This is the Kubernetes client library, and it is used for interacting with the Kubernetes API. It is critical for managing resources like secrets, config maps, and clusters. The code in `pkg` and `internal` uses this library to interact with a Kubernetes cluster, fetch resources, and apply changes based on the configurations provided.
+
+These libraries are essential for handling command-line interfaces, logging, validation, templating, and Kubernetes interactions. Their usage is relatively intuitive, and you can see their application by following the relevant sections in the codebase. For instance, `cobra` is mainly in `cmd`, `logrus` is used for error and event logging throughout, and `jsonschema` is utilized in validating the configuration YAML.
+
+</details>
+
+---
+
+### **How does caching work, and where can the caching behavior be modified in the code?**
+
+<details>
+<summary>Answer</summary>
+
+Caching is implemented directly in the file download process. Every file downloaded from an external source is saved in the local cache directory, which resides within the project’s configuration folder. The cache helps avoid downloading the same files again, improving performance and reducing reliance on external connections.
+
+The code that handles this functionality can be found in `pkg/dependencies/download.go` at line 43, where the files are downloaded, and in `pkg/x/net/client.go` at line 65, where caching is managed. If you want to modify the caching behavior, you can intervene on these files to add custom logic, such as version validation or timestamp checks to determine when to update the cache.
+
+</details>
+
+---
+
+### **How do you use `furyctl` in airgapped environments, and what flags are useful in these cases?**
+
+<details>
+<summary>Answer</summary>
+
+In airgapped environments, where no external connection is available to download binaries and dependencies, everything must be pre-downloaded and available locally. Binaries and resources can be copied manually into the `.furyctl` folder via tools like Ansible or committed directly to the project.
+
+To use `furyctl` in these environments, the following flags should be used:
+
+- `--distro-location`: This flag specifies the local path of the downloaded distribution, allowing `furyctl` to use the local version instead of attempting to download it.
+- `--skip-deps-download`: This flag skips downloading additional dependencies or binaries from external sources, ensuring that everything is used from the cache or the local distribution.
```

Comment:
Can we add a real world example? Maybe ask @stefanoghinelli with the example commands and the directory structure

Reply:
we have some docs here: https://docs.kubernetesfury.com/docs/advanced-use-cases/air-gapped